core/src/main/java/ch/cyberduck/core/vault/DecryptingListProgressListener.java

```diff
@@ -16,33 +16,50 @@
  */
 
 import ch.cyberduck.core.AttributedList;
+import ch.cyberduck.core.Filter;
 import ch.cyberduck.core.IndexedListProgressListener;
 import ch.cyberduck.core.ListProgressListener;
 import ch.cyberduck.core.Path;
 import ch.cyberduck.core.Session;
 import ch.cyberduck.core.exception.BackgroundException;
 import ch.cyberduck.core.exception.ConnectionCanceledException;
 import ch.cyberduck.core.features.Vault;
+import ch.cyberduck.core.preferences.PreferencesFactory;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.regex.Pattern;
+
 public class DecryptingListProgressListener extends IndexedListProgressListener {
     private static final Logger log = LogManager.getLogger(DecryptingListProgressListener.class);
 
     private final Session<?> session;
     private final Vault vault;
     private final ListProgressListener delegate;
+    private final Filter<Path> skip;
 
     public DecryptingListProgressListener(final Session<?> session, final Vault vault, final ListProgressListener delegate) {
+        this(session, vault, delegate, new SkipRegexFilter(Pattern.compile(PreferencesFactory.get().getProperty("cryptomator.vault.skip.regex"))));
+    }
+
+    public DecryptingListProgressListener(final Session<?> session, final Vault vault, final ListProgressListener delegate, final Filter<Path> skip) {
         this.session = session;
         this.vault = vault;
         this.delegate = delegate;
+        this.skip = skip;
     }
 
     @Override
     public void visit(final AttributedList<Path> list, final int index, final Path f) {
         try {
+            if(!skip.accept(f)) {
```

Comment:
Would probably be better to inverse the logic and skip on `true`. Thus make the filter return `true` when the skip pattern matches.